File: src/python/plugins/core/pkg/foundationallm/models/orchestration/completion_request_base.py
from typing import List, Optional
from pydantic import BaseModel, Field
from foundationallm.models.attachments import AttachmentProperties
from foundationallm.models.messages import MessageHistoryItem
from .file_history_item import FileHistoryItem


class ImageGenerationConfig(BaseModel):
    """Configuration for image generation in completion requests."""
    aspect_ratio: Optional[str] = Field("1:1", description="Aspect ratio for generated images (e.g., '1:1', '16:9', '9:16', '4:3', '3:4')")
    response_modalities: Optional[List[str]] = Field(["IMAGE"], description="Response modalities: ['IMAGE'], ['TEXT', 'IMAGE'], etc.")


class CompletionRequestBase(BaseModel):
    """
    Base class for completion requests.
    """
    operation_id: str = Field(description="The operation ID for the completion request.")
    session_id: Optional[str] = Field(None, description="The session ID for the completion request.")
    user_prompt: str = Field(description="The user prompt for the completion request.")
    user_prompt_rewrite: Optional[str] = Field(None, description="The user prompt rewrite for the completion request.")
    message_history: Optional[List[MessageHistoryItem]] = Field(list, description="The message history for the completion.")
    file_history: Optional[List[FileHistoryItem]] = Field(list, description="The file history for the completion request.")
    attachments: Optional[List[AttachmentProperties]] = Field(list, description="The attachments collection for the completion request.")
    image_config: Optional[ImageGenerationConfig] = Field(None, description="Optional image generation configuration for this request.")

File: src/dotnet/Common/Models/Orchestration/Request/CompletionRequestBase.cs
using FoundationaLLM.Common.Models.Conversation;
using System.Text.Json.Serialization;

namespace FoundationaLLM.Common.Models.Orchestration.Request
{
    /// <summary>
    /// Base for completion requests
    /// </summary>
    public class CompletionRequestBase
    {
        /// <summary>
        /// Gets or sets the operation identifier of the completion request.
        /// </summary>
        [JsonPropertyName("operation_id")]
        public string? OperationId { get; set; }

        /// <summary>
        /// Gets or sets a flag that indicates whether this is a long-running operation.
        /// </summary>
        [JsonPropertyName("long_running_operation")]
        public bool LongRunningOperation { get; set; }

        /// <summary>
        /// Gets or sets the conversation identifier.
        /// </summary>
        [JsonPropertyName("session_id")]
        public string? SessionId { get; set; }

        /// <summary>
        /// Gets or sets a value indicating whether this is a new conversation.
        /// </summary>
        [JsonPropertyName("is_new_conversation")]
        public bool IsNewConversation { get; set; }

        /// <summary>
        /// Gets or sets the user prompt.
        /// </summary>
        [JsonPropertyName("user_prompt")]
        public required string UserPrompt { get; set; }

        /// <summary>
        /// Gets or sets the rewrite of the user prompt.
        /// </summary>
        [JsonPropertyName("user_prompt_rewrite")]
        public string? UserPromptRewrite { get; set; }

        /// <summary>
        /// Gets or sets the message history associated with the completion request.
        /// </summary>
        [JsonPropertyName("message_history")]
        public List<MessageHistoryItem>? MessageHistory { get; set; } = [];

        /// <summary>
        /// Gets or sets the file history associated with the completion request.
        /// </summary>
        [JsonPropertyName("file_history")]
        public List<FileHistoryItem> FileHistory { get; set; } = [];

        /// <summary>
        /// Gets or sets the metadata associated with the completion request.
        /// </summary>
        [JsonPropertyName("metadata")]
        public Dictionary<string, object>? Metadata { get; set; }
    }
}
